app/components/Sidebar/components/CollectionLink.tsx

```diff
@@ -131,17 +131,18 @@ const CollectionLink: React.FC<Props> = ({
         <SidebarLink
           to={{
             pathname: collection.path,
-            state: { starred: inStarredSection },
+            state: { sidebarContext },
           }}
           expanded={expanded}
           onDisclosureClick={onDisclosureClick}
           onClickIntent={handlePrefetch}
           icon={<CollectionIcon collection={collection} expanded={expanded} />}
           showActions={menuOpen}
           isActiveDrop={isOver && canDrop}
-          isActive={(match, location: Location<{ starred?: boolean }>) =>
-            !!match && location.state?.starred === inStarredSection
-          }
+          isActive={(
+            match,
+            location: Location<{ sidebarContext?: SidebarContextType }>
+          ) => !!match && location.state?.sidebarContext === sidebarContext}
```

Comment:
TLDR: Instead of multiple hooks and things like `starred` just compare `sidebarContext` to see where in the sidebar the current component is being rendered